### PR TITLE
[SPARK][Docs] Added note on meaning of position to substring function

### PR DESCRIPTION
**Pull request by maclockard**
## What changes were proposed in this pull request?

Enhanced some existing documentation

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Comment by srowen** [comment]
Update `functions.py` too? I think this then matches the documentation for the function in stringExpressions.

**Comment by srowen** [comment]
Ping @maclockard ?

**Comment by srowen** [comment]
@maclockard do you want to update the other string?

**Comment by maclockard** [comment]
@srowen updated

**Review by srowen — APPROVED** [review]

File: python/pyspark/sql/functions.py [hunk 1]
```diff
@@ -1424,7 +1424,9 @@ def substring(str, pos, len):
     """
     Substring starts at `pos` and is of length `len` when str is String type or
     returns the slice of byte array that starts at `pos` in byte and is of length `len`
-    when str is Binary type
+    when str is Binary type.
+
```
Comment (srowen) [inline python/pyspark/sql/functions.py]: Ah, PEP8 doens't like the trailing whitespace:
`./python/pyspark/sql/functions.py:1428:1: W293 blank line contains whitespace`

**Comment by srowen** [comment]
Ping @maclockard let's get this one finished

**Comment by HyukjinKwon** [comment]
gentle ping @maclockard.

**Comment by maclockard** [comment]
updated

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**